components/cli/vendor/github.com/docker/docker/registry/config_unix.go

```diff
@@ -21,5 +21,5 @@ func cleanPath(s string) string {
 
 // installCliPlatformFlags handles any platform specific flags for the service.
 func (options *ServiceOptions) installCliPlatformFlags(flags *pflag.FlagSet) {
-	flags.BoolVar(&options.V2Only, "disable-legacy-registry", false, "Disable contacting legacy registries")
+	flags.BoolVar(&options.V2Only, "disable-legacy-registry", true, "Disable contacting legacy registries")
```

Comment:
cause of this discrepancy is same reason as https://github.com/docker/docker-ce/pull/116#discussion_r130778764: PR #71 should have had another commit to update `components/cli/vendor`